content/outcomes/secure-software-supply-chain/building-a-container.md

```diff
@@ -0,0 +1,91 @@
+docker build -t USERNAME/my-application .
+```
+
+However, Gradle would place the build artifact in the `build/libs` directory. Luckily, the `docker build` command allows up to override this `ARG` instruction in the command line:
+
+```
+docker build --build-arg JAR_FILE=build/libs/\*.jar -t USERNAME/my-application .
+```
+
+There's many more instructions that a Dockerfile can contain. To learn more, please refer to the [Dockerfile reference](https://docs.docker.com/engine/reference/builder/) and the [best practices for writing Dockerfiles](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/).
+
+## Kaniko
+
+[Kaniko](https://github.com/GoogleContainerTools/kaniko) is an example of a tool that can help formalize your build pipeline. Where building a Dockerfile directly requires you to have Docker up and running wherever you're performing your build, Kaniko allows you to build a container from a Dockerfile on top of Kubernetes. At its core, Kaniko is a prebuilt container image with an executable that knows how to compile a Dockerfile without Docker. You still write your Dockerfile as demonstrated, but instead of running a `docker build` command, you provide it to Kaniko along with your code, and it will build and upload the container wherever you specify.
+
+Consider the following Kubernetes pod definition:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kaniko
+spec:
+  containers:
+  - name: kaniko
+    image: gcr.io/kaniko-project/executor:latest
+    args: ["--dockerfile=Dockerfile",
+            "--context=git://github.com/ORG/REPO.git#refs/heads/main",
+            "--destination==<DOCKER USERNAME>/<CONTAINER NAME>"]
+    volumeMounts:
+      - name: kaniko-secret
+        mountPath: /kaniko/.docker
+  restartPolicy: Never
+  volumes:
+    - name: kaniko-secret
+      secret:
+        secretName: regcred
+        items:
+          - key: .dockerconfigjson
+            path: config.json
+```
+
+In this case, a pod will be spun up in Kubernetes using the Kaniko executor image, which will pull down the latest commit in the `main` branch of the GitHub repository defined in the `--context` argument, build it using the Dockerfile in the repository, and upload it to the container registry defined in the `--destination` argument. This build process, and the build process of other similar tools, makes it much easier to integrate a container build into your CI/CD pipelines.
+
+To learn more about Kaniko, please reference the documentation in the [Kaniko GitHub repository](https://github.com/GoogleContainerTools/kaniko).
+
+## Tanzu Build Service
+
+The final step you can take when deciding how to build your container is a fully automated build service such as [Tanzu Build Service](https://tanzu.vmware.com/build-service).  While you can tie tools such as Kaniko into a CI/CD pipeline that you run and manage, tools such as Tanzu Build Service aim to provide that automated build-scan-publish pipeline in a single solution. 
```

Comment:
fully automated => fully-automated

Kaniko does not have scan capabilities...or even hooks that will allow you to do so. If this is necessary (it should be) then you will need to expend energy crafting a custom solution.